saleor/account/forms.py

```diff
@@ -120,10 +124,81 @@ def get_users(self, email):
         return active_users
 
     def send_mail(
-            self, subject_template_name, email_template_name, context,
-            from_email, to_email, html_email_template_name=None):
+        self,
+        subject_template_name,
+        email_template_name,
+        context,
+        from_email,
+        to_email,
+        html_email_template_name=None):
         # Passing the user object to the Celery task throws an
         # error "'User' is not JSON serializable". Since it's not used in our
         # template, we remove it from the context.
         del context['user']
         emails.send_password_reset_email.delay(context, to_email)
+
+
+class EmailChangeForm(forms.ModelForm):
+    """ Allow changing email """
+
+    user_email = forms.EmailField(
+        label=pgettext("Label of the user's email field", "User Email"))
+
+    error_messages = {
+        'requests_exceeded':
+        pgettext_lazy(
+            "Max requests exceeded",
+            'Sorry, but it looks like You are trying change email address'
+            'again. You can change email only once per 10 minutes.'),
+        'same_email':
+        pgettext_lazy(
+            "Email can't be the same", 'New email address cannot be the '
+            'same as your current email'
+            ' address'),
+        'user_exists':
+        pgettext_lazy(
+            "User already exists",
+            'User with this email address already exists'), }
+
+    class Meta:
+        model = User
+        fields = ['user_email']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['user_email'].widget.attrs[
+            'placeholder'] = self.instance.email
+
+    def clean(self):
+        last_request = self.instance.email_change_requested_on
+        data = self.cleaned_data
+        if self.instance.email == data['user_email']:
+            self.add_error('user_email', self.error_messages['same_email'])
+        elif last_request and (last_request + timedelta(minutes=10) >
+                               timezone.now()):
+            self.add_error(
+                'user_email', self.error_messages['requests_exceeded'])
+        else:
+            try:
+                user = User.objects.get(email=data['user_email'])
```

Comment:
You might want to take a look at [exists](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#django.db.models.query.QuerySet.exists) queryset method